Frontend_Web_Development_React_Angular_Vue/Vue/Week 2/Data_Manipulation_using_Methods.md

```diff
@@ -0,0 +1,195 @@
+
+```js
+let response = [
+    {
+        mood: "happy",
+        fish: "robin",
+        colors: ["blue", "green"]
+    },
+    {
+        mood: "tired",
+        fish: "panther",
+        colors: ["green", "black", "orange", "blue"]
+    },
+    {
+        mood: "sad",
+        fish: "goldfish",
+        colors: ["green", "red"]
+    }
+]
+let myColors = [];
+for(let i=0; i<response.length; i++) {
+    myColors.push(response[i].colors)
+}
+```
+
+By the above code, you would get all the colors in the _myColors_ variable and you can then manipulate and use it accordingly.
+
+So here the need for data manipulation arises, you just don't want raw data coming from an endpoint or data which is passed as props in some of the JS frameworks, you would want to use that data and display it in the format as the client wants to see it.
+
+### Using methods for data manipulation
+
+In Vue.js to declare custom methods or to write a logic that should run at a particular instance, we write them in the methods object provided by the VUe.js instance. The **this** value is automatically bound to **methods** by Vue.js such that each function in the method has its unique copy of the component instance. Here **this** refers to the component instance. 
+
+With the help of **this** keyword, we can access the data properties in the component and manipulate them when any event listener is fired and that particular event matches with that in the methods object.
+
+Just like any other property of Vue component methods are accessible within the component's template. The most common way of using methods in a template is as follows:
+
+```html
+<button @click="changebkg">Click To change background</button>
+```
+
+### Implementation of Data Manipulation using methods
+
+```html
+<template>
+  <div id="app">
+    <span>{{message}}</span>
+    <hr>
+    <button @click="changeMessage()">Click Me</button>
+  </div>
+</template>
+
+
+<script>
+export default {
+  name: "App",
+  data: () => ({
+        message: "Implementation of Data Manipulation"
+  }),
+  methods: {
+    changeMessage () {
+      this.message = "Succesfully message changed"
+    }
+  }
+};
+</script>
+````
+
+In the above code snippet you can see that in the method properties there is a function named **changeMessage()** using this we can manipulate the **message** data property and set it to the text as we want that should be displayed on the screen on clicking the button.
+
+Before Click :
+
+<div style="display: flex; justify-content: center; align-items: center">
+  <img src="./assets/Before_Click.jpg" style="width: 500px;">
+</div>
+
+After click:
+
+<div style="display: flex; justify-content: center; align-items: center">
+  <img src="./assets/After_click.jpg" style="width: 500px;">
+</div>
+<br>
+
+**Implementing Data Manipulation in Child and Parent Components**
```

Comment:
A good example buddy, but this uses custom event handling which we are going to cover in week 5.  And the other thing it is using is a concept which is child to parent data transfer. So, basically this example should not be placed here. It will be perfect to place it inside Week 6 topic 3rd.
Just save this example and explanation of this example in your local, so you can use it at that time because that topic is almost completed with your example. So with some changes, you can create a successful PR at that time.